test/unit/SourceLocator/LocatedSourceTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace BetterReflectionTest\SourceLocator;
+
+use BetterReflection\SourceLocator\LocatedSource;
+use InvalidArgumentException;
+
+/**
+ * @covers \BetterReflection\SourceLocator\LocatedSource
+ */
+class LocatedSourceTest extends \PHPUnit_Framework_TestCase
+{
+    public function testValuesHappyPath()
+    {
+        $source = '<?php echo "Hello world";';
+        $file = 'path/to/file.php';
+        $locatedSource = new LocatedSource($source, $file);
+
+        $this->assertSame($source, $locatedSource->getSource());
+        $this->assertSame($file, $locatedSource->getFileName());
+    }
+
+    public function testValuesWithNullFilename()
+    {
+        $source = '<?php echo "Hello world";';
+        $file = null;
+        $locatedSource = new LocatedSource($source, $file);
```

Comment:
For when you want to load source code directly from a string - the `StringSourceLocator` does exactly this, but the filename is allowed to be `null` (because you are loading from a string, rather than a file). It just means when you reflect from a class within a string, you'll not have a filename when you call `$classInfo->getFileName()` but then that would be expected (because you never loaded from file)